{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "2f8090a6",
   "metadata": {},
   "outputs": [],
   "source": [
    "netid = \"kolodisner\"\n",
    "xfile = \"xac\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "1fbc81e9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses\n"
     ]
    }
   ],
   "source": [
    "# go into the working directory \n",
    "work_dir = \"/xdisk/bhurwitz/virus_hunting/\" + netid + \"/06_get_viruses\"\n",
    "%cd $work_dir "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "79b843c3",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory '/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_virsort': File exists\n",
      "mkdir: cannot create directory '/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_checkv_virsort': File exists\n",
      "mkdir: cannot create directory '/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_dvf': File exists\n",
      "mkdir: cannot create directory '/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_checkv_dvf': File exists\n",
      "mkdir: cannot create directory '/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_genomad': File exists\n"
     ]
    }
   ],
   "source": [
    "!mkdir $work_dir/out_virsort\n",
    "!mkdir $work_dir/out_checkv_virsort\n",
    "!mkdir $work_dir/out_dvf\n",
    "!mkdir $work_dir/out_checkv_dvf\n",
    "!mkdir $work_dir/out_genomad\n",
    "!mkdir $work_dir/out_checkv_genomad"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "3f2b7ca1",
   "metadata": {},
   "outputs": [],
   "source": [
    "# make the config file \n",
    "!echo \"export NETID=$netid\" > config.sh\n",
    "!echo \"export XFILE=$xfile\" >> config.sh\n",
    "!echo \"export XFILE_DIR=/xdisk/bhurwitz/virus_hunting/$netid\" >> config.sh\n",
    "!echo \"export WORK_DIR=/xdisk/bhurwitz/virus_hunting/$netid/06_get_viruses\" >> config.sh\n",
    "!echo \"export SPADES_DIR=/xdisk/bhurwitz/virus_hunting/$netid/04_assembly/out_spades\" >> config.sh\n",
    "!echo \"export OUT_VIRSORT=/xdisk/bhurwitz/virus_hunting/$netid/06_get_viruses/out_virsort\" >> config.sh\n",
    "!echo \"export OUT_CHECKV_VIRSORT=/xdisk/bhurwitz/virus_hunting/$netid/06_get_viruses/out_checkv_virsort\" >> config.sh\n",
    "!echo \"export CHECKVDB=/groups/bhurwitz/databases/checkv-db-v1.5\" >> config.sh\n",
    "!echo \"export OUT_DVF=/xdisk/bhurwitz/virus_hunting/$netid/06_get_viruses/out_dvf\" >> config.sh\n",
    "!echo \"export DVF_DB=/groups/bhurwitz/databases/DeepVirFinder\" >> config.sh\n",
    "!echo \"export OUT_CHECKV_DVF=/xdisk/bhurwitz/virus_hunting/$netid/06_get_viruses/out_checkv_dvf\" >> config.sh\n",
    "!echo \"export OUT_GENOMAD=/xdisk/bhurwitz/virus_hunting/$netid/06_get_viruses/out_genomad\" >> config.sh\n",
    "!echo \"export GENOMAD_DB=/groups/bhurwitz/databases/genomad_db\" >> config.sh\n",
    "!echo \"export OUT_CHECKV_GENOMAD=/xdisk/bhurwitz/virus_hunting/$netid/06_get_viruses/out_checkv_genomad\" >> config.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "f686543e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "export NETID=kolodisner\r\n",
      "export XFILE=xac\r\n",
      "export XFILE_DIR=/xdisk/bhurwitz/virus_hunting/kolodisner\r\n",
      "export WORK_DIR=/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses\r\n",
      "export SPADES_DIR=/xdisk/bhurwitz/virus_hunting/kolodisner/04_assembly/out_spades\r\n",
      "export OUT_VIRSORT=/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_virsort\r\n",
      "export OUT_CHECKV_VIRSORT=/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_checkv_virsort\r\n",
      "export CHECKVDB=/groups/bhurwitz/databases/checkv-db-v1.5\r\n",
      "export OUT_DVF=/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_dvf\r\n",
      "export DVF_DB=/groups/bhurwitz/databases/DeepVirFinder\r\n",
      "export OUT_CHECKV_DVF=/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_checkv_dvf\r\n",
      "export OUT_GENOMAD=/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_genomad\r\n",
      "export GENOMAD_DB=/groups/bhurwitz/databases/genomad_db\r\n",
      "export OUT_CHECKV_GENOMAD=/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_checkv_genomad\r\n"
     ]
    }
   ],
   "source": [
    "!cat config.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "68087aaa",
   "metadata": {},
   "outputs": [],
   "source": [
    "my_code = '''#!/bin/bash\n",
    "#SBATCH --output=Job-virsort2-%a.out\n",
    "#SBATCH --account=bhurwitz\n",
    "#SBATCH --partition=standard\n",
    "#SBATCH --nodes=1\n",
    "#SBATCH --ntasks=1\n",
    "#SBATCH --time=20:00:00\n",
    "#SBATCH --cpus-per-task=4\n",
    "#SBATCH --mem-per-cpu=1000\n",
    "#SBATCH --array=0-2\n",
    "\n",
    "pwd; hostname; date\n",
    "\n",
    "source ./config.sh\n",
    "names=($(cat $XFILE_DIR/$XFILE))\n",
    "SAMPLE_ID=${names[${SLURM_ARRAY_TASK_ID}]}\n",
    "\n",
    "SPADES=${SPADES_DIR}/${SAMPLE_ID}/contigs.fasta.gz\n",
    "\n",
    "module load python/3.9/3.9.10\n",
    "module load R/4.2.2\n",
    "\n",
    "#load environment\n",
    "CONDA=\"/groups/bhurwitz/miniconda3\"\n",
    "source $CONDA/etc/profile.d/conda.sh\n",
    "conda activate viral\n",
    "\n",
    "virsorter run -w ${OUT_VIRSORT}/${names[${SLURM_ARRAY_TASK_ID}]} -i ${SPADES} --min-length 1500 -j 4 all\n",
    "'''\n",
    "\n",
    "with open('run_virsort2.sh', mode='w') as file:\n",
    "    file.write(my_code)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "0c4a40c0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses\n",
      "06_get_viruses.ipynb\t   Job-dvf-2.out       out_checkv_virsort\n",
      "Job-checkv-dvf-0.out\t   Job-genomad-0.out   out_dvf\n",
      "Job-checkv-dvf-1.out\t   Job-genomad-1.out   out_genomad\n",
      "Job-checkv-dvf-2.out\t   Job-genomad-2.out   out_virsort\n",
      "Job-checkv-genomad-0.out   Job-virsort2-0.out  run_checkv_dvf.sh\n",
      "Job-checkv-genomad-1.out   Job-virsort2-1.out  run_checkv_genomad.sh\n",
      "Job-checkv-genomad-2.out   Job-virsort2-2.out  run_checkv_virsort2.sh\n",
      "Job-checkv-virsort2-0.out  config.sh\t       run_dvf.sh\n",
      "Job-checkv-virsort2-1.out  dvf.fasta\t       run_genomad.sh\n",
      "Job-checkv-virsort2-2.out  id_from_fasta.py    run_virsort2.sh\n",
      "Job-dvf-0.out\t\t   out_checkv_dvf      temp.sh\n",
      "Job-dvf-1.out\t\t   out_checkv_genomad\n"
     ]
    }
   ],
   "source": [
    "#check that I am in working directory with config file and run_virsort2.sh script\n",
    "!pwd\n",
    "!ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "f995325f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Submitted batch job 3273655\r\n"
     ]
    }
   ],
   "source": [
    "#run script!\n",
    "!sbatch ./run_virsort2.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "d80eb563",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\r\n",
      "     3280568_[0-2]  standard run_chec kolodisn PD       0:00      1 (Priority)\r\n",
      "           3280555  standard  jupyter kolodisn  R      23:04      1 i16n5\r\n"
     ]
    }
   ],
   "source": [
    "#is the job in queue? \n",
    "!squeue --user=$netid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "c03680db",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_virsort/config.yaml\r\n",
      "\r\n",
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_virsort/ERR2198703:\r\n",
      "config.yaml\t\t  final-viral-combined.fa  iter-0\r\n",
      "final-viral-boundary.tsv  final-viral-score.tsv    log\r\n",
      "\r\n",
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_virsort/ERR2198704:\r\n",
      "config.yaml\t\t  final-viral-combined.fa  iter-0\r\n",
      "final-viral-boundary.tsv  final-viral-score.tsv    log\r\n",
      "\r\n",
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_virsort/ERR2198705:\r\n",
      "config.yaml\t\t  final-viral-combined.fa  iter-0\r\n",
      "final-viral-boundary.tsv  final-viral-score.tsv    log\r\n"
     ]
    }
   ],
   "source": [
    "#check the output files \n",
    "!ls $work_dir/out_virsort/*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "9aab5271",
   "metadata": {},
   "outputs": [],
   "source": [
    "my_code = '''#!/bin/bash\n",
    "#SBATCH --output=Job-checkv-virsort2-%a.out\n",
    "#SBATCH --account=bhurwitz\n",
    "#SBATCH --partition=standard\n",
    "#SBATCH --nodes=1\n",
    "#SBATCH --ntasks=1\n",
    "#SBATCH --time=20:00:00\n",
    "#SBATCH --cpus-per-task=4\n",
    "#SBATCH --mem-per-cpu=1000\n",
    "#SBATCH --array=0-2\n",
    "\n",
    "pwd; hostname; date\n",
    "\n",
    "source ./config.sh\n",
    "names=($(cat $XFILE_DIR/$XFILE))\n",
    "SAMPLE_ID=${names[${SLURM_ARRAY_TASK_ID}]}\n",
    "\n",
    "VIRSORT=${OUT_VIRSORT}/${SAMPLE_ID}/final-viral-combined.fa\n",
    "\n",
    "#load environment\n",
    "CONDA=\"/groups/bhurwitz/miniconda3\"\n",
    "source $CONDA/etc/profile.d/conda.sh\n",
    "conda activate checkv_env\n",
    "\n",
    "checkv end_to_end ${VIRSORT} ${OUT_CHECKV_VIRSORT}/${names[${SLURM_ARRAY_TASK_ID}]} -t 16\n",
    "'''\n",
    "\n",
    "with open('run_checkv_virsort2.sh', mode='w') as file:\n",
    "    file.write(my_code)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "ec1240aa",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses\n",
      "06_get_viruses.ipynb\t   Job-dvf-1.out       out_checkv_virsort\n",
      "DeepVirFinder\t\t   Job-dvf-2.out       out_dvf\n",
      "Job-checkv-dvf-0.out\t   Job-virsort2-0.out  out_virsort\n",
      "Job-checkv-dvf-1.out\t   Job-virsort2-1.out  run_checkv_dvf.sh\n",
      "Job-checkv-dvf-2.out\t   Job-virsort2-2.out  run_checkv_virsort2.sh\n",
      "Job-checkv-virsort2-0.out  config.sh\t       run_dvf.sh\n",
      "Job-checkv-virsort2-1.out  dvf.fasta\t       run_virsort2.sh\n",
      "Job-checkv-virsort2-2.out  id_from_fasta.py    temp.sh\n",
      "Job-dvf-0.out\t\t   out_checkv_dvf\n"
     ]
    }
   ],
   "source": [
    "#check that I am in working directory with config file and run_virsort2.sh script\n",
    "!pwd\n",
    "!ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "7c391e40",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Submitted batch job 3294546\r\n"
     ]
    }
   ],
   "source": [
    "#run script!\n",
    "!sbatch ./run_checkv_virsort2.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "a168e36b",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n",
      "         3294546_0  standard run_chec kolodisn  R       0:16      1 i16n17\n",
      "         3294546_1  standard run_chec kolodisn  R       0:16      1 i16n17\n",
      "         3294546_2  standard run_chec kolodisn  R       0:16      1 i16n19\n",
      "           3294545  standard  jupyter kolodisn  R       5:16      1 i16n12\n"
     ]
    }
   ],
   "source": [
    "#is the job in queue? \n",
    "!squeue --user=$netid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "569dd3ed",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_checkv_virsort/ERR2198703:\r\n",
      "complete_genomes.tsv  contamination.tsv  quality_summary.tsv  viruses.fna\r\n",
      "completeness.tsv      proviruses.fna\t tmp\r\n",
      "\r\n",
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_checkv_virsort/ERR2198704:\r\n",
      "complete_genomes.tsv  contamination.tsv  quality_summary.tsv  viruses.fna\r\n",
      "completeness.tsv      proviruses.fna\t tmp\r\n",
      "\r\n",
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_checkv_virsort/ERR2198705:\r\n",
      "complete_genomes.tsv  contamination.tsv  quality_summary.tsv  viruses.fna\r\n",
      "completeness.tsv      proviruses.fna\t tmp\r\n"
     ]
    }
   ],
   "source": [
    "#check the output files \n",
    "!ls $work_dir/out_checkv_virsort/*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "887333bf",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses\r\n",
      "i16n17.ocelote.hpc.arizona.edu\r\n",
      "Sat Jul 13 10:34:07 MST 2024\r\n",
      "\r\n",
      "CheckV v0.9.0: contamination\r\n",
      "[1/8] Reading database info...\r\n",
      "[2/8] Reading genome info...\r\n",
      "[3/8] Skipping gene calling...\r\n",
      "[4/8] Reading gene info...\r\n",
      "[5/8] Skipping hmmsearch...\r\n",
      "[6/8] Annotating genes...\r\n",
      "[7/8] Identifying host regions...\r\n",
      "[8/8] Writing results...\r\n",
      "Run time: 1.25 seconds\r\n",
      "Peak mem: 0.07 GB\r\n",
      "\r\n",
      "CheckV v0.9.0: completeness\r\n",
      "[1/8] Skipping gene calling...\r\n",
      "[2/8] Initializing queries and database...\r\n",
      "[3/8] Running DIAMOND blastp search...\r\n",
      "[4/8] Computing AAI...\r\n",
      "[5/8] Running AAI based completeness estimation...\r\n",
      "[6/8] Running HMM based completeness estimation...\r\n",
      "[7/8] Determining genome copy number...\r\n",
      "[8/8] Writing results...\r\n",
      "Run time: 54.99 seconds\r\n",
      "Peak mem: 1.62 GB\r\n",
      "\r\n",
      "CheckV v0.9.0: complete_genomes\r\n",
      "[1/7] Reading input sequences...\r\n",
      "[2/7] Finding complete proviruses...\r\n",
      "[3/7] Finding direct/inverted terminal repeats...\r\n",
      "[4/7] Filtering terminal repeats...\r\n",
      "[5/7] Checking genome for completeness...\r\n",
      "[6/7] Checking genome for large duplications...\r\n",
      "[7/7] Writing results...\r\n",
      "Run time: 0.07 seconds\r\n",
      "Peak mem: 1.62 GB\r\n",
      "\r\n",
      "CheckV v0.9.0: quality_summary\r\n",
      "[1/6] Reading input sequences...\r\n",
      "[2/6] Reading results from contamination module...\r\n",
      "[3/6] Reading results from completeness module...\r\n",
      "[4/6] Reading results from complete genomes module...\r\n",
      "[5/6] Classifying contigs into quality tiers...\r\n",
      "[6/6] Writing results...\r\n",
      "Run time: 0.04 seconds\r\n",
      "Peak mem: 1.62 GB\r\n",
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses\r\n",
      "i16n17.ocelote.hpc.arizona.edu\r\n",
      "Sat Jul 13 10:34:07 MST 2024\r\n",
      "\r\n",
      "CheckV v0.9.0: contamination\r\n",
      "[1/8] Reading database info...\r\n",
      "[2/8] Reading genome info...\r\n",
      "[3/8] Skipping gene calling...\r\n",
      "[4/8] Reading gene info...\r\n",
      "[5/8] Skipping hmmsearch...\r\n",
      "[6/8] Annotating genes...\r\n",
      "[7/8] Identifying host regions...\r\n",
      "[8/8] Writing results...\r\n",
      "Run time: 1.63 seconds\r\n",
      "Peak mem: 0.08 GB\r\n",
      "\r\n",
      "CheckV v0.9.0: completeness\r\n",
      "[1/8] Skipping gene calling...\r\n",
      "[2/8] Initializing queries and database...\r\n",
      "[3/8] Running DIAMOND blastp search...\r\n",
      "[4/8] Computing AAI...\r\n",
      "[5/8] Running AAI based completeness estimation...\r\n",
      "[6/8] Running HMM based completeness estimation...\r\n",
      "[7/8] Determining genome copy number...\r\n",
      "[8/8] Writing results...\r\n",
      "Run time: 58.45 seconds\r\n",
      "Peak mem: 1.54 GB\r\n",
      "\r\n",
      "CheckV v0.9.0: complete_genomes\r\n",
      "[1/7] Reading input sequences...\r\n",
      "[2/7] Finding complete proviruses...\r\n",
      "[3/7] Finding direct/inverted terminal repeats...\r\n",
      "[4/7] Filtering terminal repeats...\r\n",
      "[5/7] Checking genome for completeness...\r\n",
      "[6/7] Checking genome for large duplications...\r\n",
      "[7/7] Writing results...\r\n",
      "Run time: 0.13 seconds\r\n",
      "Peak mem: 1.54 GB\r\n",
      "\r\n",
      "CheckV v0.9.0: quality_summary\r\n",
      "[1/6] Reading input sequences...\r\n",
      "[2/6] Reading results from contamination module...\r\n",
      "[3/6] Reading results from completeness module...\r\n",
      "[4/6] Reading results from complete genomes module...\r\n",
      "[5/6] Classifying contigs into quality tiers...\r\n",
      "[6/6] Writing results...\r\n",
      "Run time: 0.06 seconds\r\n",
      "Peak mem: 1.54 GB\r\n",
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses\r\n",
      "i16n19.ocelote.hpc.arizona.edu\r\n",
      "Sat Jul 13 10:34:07 MST 2024\r\n",
      "\r\n",
      "CheckV v0.9.0: contamination\r\n",
      "[1/8] Reading database info...\r\n",
      "[2/8] Reading genome info...\r\n",
      "[3/8] Skipping gene calling...\r\n",
      "[4/8] Reading gene info...\r\n",
      "[5/8] Skipping hmmsearch...\r\n",
      "[6/8] Annotating genes...\r\n",
      "[7/8] Identifying host regions...\r\n",
      "[8/8] Writing results...\r\n",
      "Run time: 1.64 seconds\r\n",
      "Peak mem: 0.08 GB\r\n",
      "\r\n",
      "CheckV v0.9.0: completeness\r\n",
      "[1/8] Skipping gene calling...\r\n",
      "[2/8] Initializing queries and database...\r\n",
      "[3/8] Running DIAMOND blastp search...\r\n",
      "[4/8] Computing AAI...\r\n",
      "[5/8] Running AAI based completeness estimation...\r\n",
      "[6/8] Running HMM based completeness estimation...\r\n",
      "[7/8] Determining genome copy number...\r\n",
      "[8/8] Writing results...\r\n",
      "Run time: 58.45 seconds\r\n",
      "Peak mem: 1.52 GB\r\n",
      "\r\n",
      "CheckV v0.9.0: complete_genomes\r\n",
      "[1/7] Reading input sequences...\r\n",
      "[2/7] Finding complete proviruses...\r\n",
      "[3/7] Finding direct/inverted terminal repeats...\r\n",
      "[4/7] Filtering terminal repeats...\r\n",
      "[5/7] Checking genome for completeness...\r\n",
      "[6/7] Checking genome for large duplications...\r\n",
      "[7/7] Writing results...\r\n",
      "Run time: 0.12 seconds\r\n",
      "Peak mem: 1.52 GB\r\n",
      "\r\n",
      "CheckV v0.9.0: quality_summary\r\n",
      "[1/6] Reading input sequences...\r\n",
      "[2/6] Reading results from contamination module...\r\n",
      "[3/6] Reading results from completeness module...\r\n",
      "[4/6] Reading results from complete genomes module...\r\n",
      "[5/6] Classifying contigs into quality tiers...\r\n",
      "[6/6] Writing results...\r\n",
      "Run time: 0.06 seconds\r\n",
      "Peak mem: 1.52 GB\r\n"
     ]
    }
   ],
   "source": [
    "!cat Job-checkv-virsort2-*.out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "3b174e5f",
   "metadata": {},
   "outputs": [],
   "source": [
    "my_code = '''#!/bin/bash\n",
    "#SBATCH --output=Job-dvf-%a.out\n",
    "#SBATCH --account=bhurwitz\n",
    "#SBATCH --partition=standard\n",
    "#SBATCH --nodes=1\n",
    "#SBATCH --ntasks=1\n",
    "#SBATCH --time=20:00:00\n",
    "#SBATCH --cpus-per-task=4\n",
    "#SBATCH --mem-per-cpu=5000\n",
    "#SBATCH --array=0-2\n",
    "\n",
    "pwd; hostname; date\n",
    "\n",
    "source ./config.sh\n",
    "names=($(cat $XFILE_DIR/$XFILE))\n",
    "SAMPLE_ID=${names[${SLURM_ARRAY_TASK_ID}]}\n",
    "\n",
    "SPADES=${SPADES_DIR}/${SAMPLE_ID}/contigs.fasta.gz\n",
    "\n",
    "#load environment\n",
    "CONDA=\"/groups/bhurwitz/miniconda3\"\n",
    "source $CONDA/etc/profile.d/conda.sh\n",
    "conda activate dvf\n",
    "\n",
    "cd ${DVF_DB}\n",
    "\n",
    "python dvf.py -i ${SPADES} -o ${OUT_DVF}/${names[${SLURM_ARRAY_TASK_ID}]} -l 1500\n",
    "\n",
    "cd ${WORK_DIR}\n",
    "'''\n",
    "\n",
    "with open('run_dvf.sh', mode='w') as file:\n",
    "    file.write(my_code)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "4c1eb6ce",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses\n",
      "06_get_viruses.ipynb\t   Job-genomad-1.out   out_dvf\n",
      "Job-checkv-dvf-0.out\t   Job-genomad-2.out   out_genomad\n",
      "Job-checkv-dvf-1.out\t   Job-virsort2-0.out  out_virsort\n",
      "Job-checkv-dvf-2.out\t   Job-virsort2-1.out  run_checkv_dvf.sh\n",
      "Job-checkv-virsort2-0.out  Job-virsort2-2.out  run_checkv_genomad.sh\n",
      "Job-checkv-virsort2-1.out  config.sh\t       run_checkv_virsort2.sh\n",
      "Job-checkv-virsort2-2.out  dvf.fasta\t       run_dvf.sh\n",
      "Job-dvf-0.out\t\t   id_from_fasta.py    run_genomad.sh\n",
      "Job-dvf-1.out\t\t   out_checkv_dvf      run_virsort2.sh\n",
      "Job-dvf-2.out\t\t   out_checkv_genomad  temp.sh\n",
      "Job-genomad-0.out\t   out_checkv_virsort\n"
     ]
    }
   ],
   "source": [
    "!pwd\n",
    "!ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "a8a27eb3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Submitted batch job 3298559\r\n"
     ]
    }
   ],
   "source": [
    "!sbatch ./run_dvf.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "a89f3f29",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n",
      "         3298559_0  standard run_dvf. kolodisn  R       5:22      1 i16n1\n",
      "         3298559_1  standard run_dvf. kolodisn  R       5:22      1 i16n3\n",
      "         3298559_2  standard run_dvf. kolodisn  R       5:22      1 i16n3\n",
      "         3298554_1  standard run_geno kolodisn  R      15:22      1 i16n2\n",
      "         3298554_2  standard run_geno kolodisn  R      15:22      1 i16n2\n",
      "           3298549  standard  jupyter kolodisn  R    1:10:25      1 i16n0\n"
     ]
    }
   ],
   "source": [
    "!squeue --user=$netid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "e63b3916",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_dvf/ERR2198703:\r\n",
      "contigs.fasta_gt1500bp_dvfpred.txt  dvf.fasta\r\n",
      "\r\n",
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_dvf/ERR2198704:\r\n",
      "contigs.fasta_gt1500bp_dvfpred.txt  dvf.fasta\r\n",
      "\r\n",
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_dvf/ERR2198705:\r\n",
      "contigs.fasta_gt1500bp_dvfpred.txt  dvf.fasta\r\n"
     ]
    }
   ],
   "source": [
    "!ls $work_dir/out_dvf/*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "6673a9fc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses\r\n",
      "i16n1.ocelote.hpc.arizona.edu\r\n",
      "Sat Jul 20 20:54:05 MST 2024\r\n",
      "WARNING (theano.tensor.blas): Using NumPy C-API based implementation for BLAS functions.\r\n",
      "Using Theano backend.\r\n",
      "INFO (theano.gof.compilelock): Waiting for existing lock by unknown process (I am process '28262')\r\n",
      "INFO (theano.gof.compilelock): To manually release the lock, delete /home/u3/kolodisner/.theano/compiledir_Linux-3.10-el7.x86_64-x86_64-with-centos-7.9.2009-Core-x86_64-3.6.13-64/lock_dir\r\n",
      "1. Loading Models.\r\n",
      "   model directory /groups/bhurwitz/databases/DeepVirFinder/models\r\n",
      "2. Encoding and Predicting Sequences.\r\n",
      "   processing line 1\r\n",
      "   processing line 370972\r\n",
      "   processing line 566554\r\n",
      "   processing line 708605\r\n",
      "   processing line 819212\r\n",
      "   processing line 911178\r\n",
      "   processing line 987504\r\n",
      "   processing line 1051652\r\n",
      "   processing line 1107397\r\n",
      "   processing line 1156201\r\n",
      "   processing line 1200214\r\n",
      "   processing line 1239469\r\n",
      "   processing line 1275085\r\n",
      "   processing line 1307163\r\n",
      "   processing line 1336287\r\n",
      "   processing line 1363389\r\n",
      "   processing line 1388675\r\n",
      "   processing line 1412322\r\n",
      "   processing line 1434382\r\n",
      "   processing line 1455217\r\n",
      "   processing line 1474949\r\n",
      "   processing line 1493633\r\n",
      "   processing line 1511339\r\n",
      "   processing line 1528204\r\n",
      "   processing line 1544254\r\n",
      "   processing line 1559655\r\n",
      "   processing line 1574445\r\n",
      "   processing line 1588720\r\n",
      "   processing line 1602416\r\n",
      "   processing line 1615662\r\n",
      "   processing line 1628452\r\n",
      "   processing line 1640821\r\n",
      "   processing line 1652811\r\n",
      "   processing line 1664392\r\n",
      "   processing line 1675616\r\n",
      "   processing line 1686540\r\n",
      "   processing line 1697137\r\n",
      "   processing line 1707425\r\n",
      "   processing line 1717454\r\n",
      "   processing line 1727248\r\n",
      "   processing line 1736811\r\n",
      "   processing line 1746125\r\n",
      "   processing line 1755211\r\n",
      "   processing line 1764120\r\n",
      "   processing line 1772852\r\n",
      "   processing line 1781406\r\n",
      "   processing line 1789772\r\n",
      "   processing line 1797995\r\n",
      "   processing line 1806047\r\n",
      "   processing line 1813960\r\n",
      "   processing line 1821726\r\n",
      "   processing line 1829367\r\n",
      "   processing line 1836872\r\n",
      "   processing line 1844255\r\n",
      "   processing line 1851518\r\n",
      "   processing line 1858674\r\n",
      "   processing line 1865690\r\n",
      "   processing line 1872579\r\n",
      "   processing line 1879361\r\n",
      "   processing line 1886020\r\n",
      "   processing line 1892570\r\n",
      "   processing line 1899024\r\n",
      "   processing line 1905381\r\n",
      "   processing line 1911648\r\n",
      "   processing line 1917829\r\n",
      "   processing line 1923929\r\n",
      "   processing line 1929949\r\n",
      "   processing line 1935876\r\n",
      "   processing line 1941738\r\n",
      "   processing line 1947516\r\n",
      "   processing line 1953216\r\n",
      "   processing line 1958819\r\n",
      "   processing line 1964350\r\n",
      "   processing line 1969834\r\n",
      "   processing line 1975234\r\n",
      "   processing line 1980554\r\n",
      "   processing line 1985832\r\n",
      "   processing line 1991032\r\n",
      "   processing line 1996172\r\n",
      "   processing line 2001262\r\n",
      "   processing line 2006262\r\n",
      "   processing line 2011219\r\n",
      "   processing line 2016119\r\n",
      "   processing line 2020967\r\n",
      "   processing line 2025767\r\n",
      "   processing line 2030505\r\n",
      "   processing line 2035205\r\n",
      "   processing line 2039833\r\n",
      "   processing line 2044433\r\n",
      "   processing line 2048984\r\n",
      "   processing line 2053484\r\n",
      "   processing line 2057970\r\n",
      "   processing line 2062370\r\n",
      "   processing line 2066770\r\n",
      "   processing line 2071090\r\n",
      "   processing line 2075390\r\n",
      "   processing line 2079672\r\n",
      "   processing line 2083872\r\n",
      "   processing line 2088072\r\n",
      "   processing line 2092184\r\n",
      "   processing line 2096284\r\n",
      "   processing line 2100381\r\n",
      "   processing line 2104381\r\n",
      "   processing line 2108381\r\n",
      "   processing line 2112364\r\n",
      "   processing line 2116264\r\n",
      "   processing line 2120164\r\n",
      "   processing line 2124023\r\n",
      "   processing line 2127823\r\n",
      "   processing line 2131623\r\n",
      "   processing line 2135379\r\n",
      "   processing line 2139079\r\n",
      "   processing line 2142779\r\n",
      "   processing line 2146472\r\n",
      "   processing line 2150072\r\n",
      "   processing line 2153672\r\n",
      "   processing line 2157272\r\n",
      "   processing line 2160814\r\n",
      "   processing line 2164314\r\n",
      "   processing line 2167814\r\n",
      "   processing line 2171314\r\n",
      "   processing line 2174761\r\n",
      "   processing line 2178161\r\n",
      "   processing line 2181561\r\n",
      "   processing line 2184961\r\n",
      "   processing line 2188320\r\n",
      "   processing line 2191620\r\n",
      "   processing line 2194920\r\n",
      "   processing line 2198220\r\n",
      "   processing line 2201516\r\n",
      "   processing line 2204716\r\n",
      "   processing line 2207916\r\n",
      "   processing line 2211116\r\n",
      "   processing line 2214316\r\n",
      "   processing line 2217483\r\n",
      "   processing line 2220583\r\n",
      "   processing line 2223683\r\n",
      "   processing line 2226783\r\n",
      "   processing line 2229883\r\n",
      "   processing line 2232983\r\n",
      "   processing line 2236011\r\n",
      "   processing line 2239011\r\n",
      "   processing line 2242011\r\n",
      "   processing line 2245011\r\n",
      "   processing line 2248011\r\n",
      "   processing line 2250966\r\n",
      "   processing line 2253866\r\n",
      "   processing line 2256766\r\n",
      "   processing line 2259666\r\n",
      "   processing line 2262566\r\n",
      "   processing line 2265466\r\n",
      "   processing line 2268286\r\n",
      "   processing line 2271086\r\n",
      "   processing line 2273886\r\n",
      "   processing line 2276686\r\n",
      "   processing line 2279486\r\n",
      "   processing line 2282286\r\n",
      "   processing line 2285008\r\n",
      "   processing line 2287708\r\n",
      "   processing line 2290408\r\n",
      "   processing line 2293108\r\n",
      "   processing line 2295808\r\n",
      "   processing line 2298508\r\n",
      "   processing line 2301208\r\n",
      "   processing line 3630687\r\n",
      "3. Done. Thank you for using DeepVirFinder.\r\n",
      "   output in /xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_dvf/ERR2198703/contigs.fasta_gt1500bp_dvfpred.txt\r\n",
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses\r\n",
      "i16n3.ocelote.hpc.arizona.edu\r\n",
      "Sat Jul 20 20:54:06 MST 2024\r\n",
      "WARNING (theano.tensor.blas): Using NumPy C-API based implementation for BLAS functions.\r\n",
      "1. Loading Models.\r\n",
      "   model directory /groups/bhurwitz/databases/DeepVirFinder/models\r\n",
      "2. Encoding and Predicting Sequences.\r\n",
      "   processing line 1\r\n",
      "   processing line 316102\r\n",
      "Using Theano backend.\r\n",
      "   processing line 469146\r\n",
      "   processing line 583508\r\n",
      "   processing line 678851\r\n",
      "   processing line 761589\r\n",
      "   processing line 834295\r\n",
      "   processing line 899197\r\n",
      "   processing line 958516\r\n",
      "   processing line 1012684\r\n",
      "   processing line 1062408\r\n",
      "   processing line 1108706\r\n",
      "   processing line 1152254\r\n",
      "   processing line 1193352\r\n",
      "   processing line 1231904\r\n",
      "   processing line 1268222\r\n",
      "   processing line 1302374\r\n",
      "   processing line 1334667\r\n",
      "   processing line 1365563\r\n",
      "   processing line 1394994\r\n",
      "   processing line 1423152\r\n",
      "   processing line 1450260\r\n",
      "   processing line 1476424\r\n",
      "   processing line 1501578\r\n",
      "   processing line 1525788\r\n",
      "   processing line 1549056\r\n",
      "   processing line 1571508\r\n",
      "   processing line 1593269\r\n",
      "   processing line 1614388\r\n",
      "   processing line 1634911\r\n",
      "   processing line 1654857\r\n",
      "   processing line 1674339\r\n",
      "   processing line 1693262\r\n",
      "   processing line 1711641\r\n",
      "   processing line 1729581\r\n",
      "   processing line 1747050\r\n",
      "   processing line 1764068\r\n",
      "   processing line 1780659\r\n",
      "   processing line 1796902\r\n",
      "   processing line 1812773\r\n",
      "   processing line 1828305\r\n",
      "   processing line 1843520\r\n",
      "   processing line 1858371\r\n",
      "   processing line 1872938\r\n",
      "   processing line 1887205\r\n",
      "   processing line 1901163\r\n",
      "   processing line 1914860\r\n",
      "   processing line 1928316\r\n",
      "   processing line 1941552\r\n",
      "   processing line 1954537\r\n",
      "   processing line 1967270\r\n",
      "   processing line 1979758\r\n",
      "   processing line 1991957\r\n",
      "   processing line 2003942\r\n",
      "   processing line 2015754\r\n",
      "   processing line 2027400\r\n",
      "   processing line 2038864\r\n",
      "   processing line 2050166\r\n",
      "   processing line 2061313\r\n",
      "   processing line 2072268\r\n",
      "   processing line 2083056\r\n",
      "   processing line 2093687\r\n",
      "   processing line 2104177\r\n",
      "   processing line 2114520\r\n",
      "   processing line 2124733\r\n",
      "   processing line 2134800\r\n",
      "   processing line 2144746\r\n",
      "   processing line 2154570\r\n",
      "   processing line 2164269\r\n",
      "   processing line 2173851\r\n",
      "   processing line 2183325\r\n",
      "   processing line 2192683\r\n",
      "   processing line 2201932\r\n",
      "   processing line 2211066\r\n",
      "   processing line 2220098\r\n",
      "   processing line 2229023\r\n",
      "   processing line 2237855\r\n",
      "   processing line 2246577\r\n",
      "   processing line 2255204\r\n",
      "   processing line 2263728\r\n",
      "   processing line 2272156\r\n",
      "   processing line 2280509\r\n",
      "   processing line 2288785\r\n",
      "   processing line 2296977\r\n",
      "   processing line 2305077\r\n",
      "   processing line 2313090\r\n",
      "   processing line 2321033\r\n",
      "   processing line 2328884\r\n",
      "   processing line 2336659\r\n",
      "   processing line 2344359\r\n",
      "   processing line 2351960\r\n",
      "   processing line 2359497\r\n",
      "   processing line 2366969\r\n",
      "   processing line 2374369\r\n",
      "   processing line 2381678\r\n",
      "   processing line 2388926\r\n",
      "   processing line 2396123\r\n",
      "   processing line 2403223\r\n",
      "   processing line 2410265\r\n",
      "   processing line 2417265\r\n",
      "   processing line 2424167\r\n",
      "   processing line 2431036\r\n",
      "   processing line 2437836\r\n",
      "   processing line 2444577\r\n",
      "   processing line 2451247\r\n",
      "   processing line 2457847\r\n",
      "   processing line 2464387\r\n",
      "   processing line 2470887\r\n",
      "   processing line 2477327\r\n",
      "   processing line 2483727\r\n",
      "   processing line 2490073\r\n",
      "   processing line 2496373\r\n",
      "   processing line 2502614\r\n",
      "   processing line 2508814\r\n",
      "   processing line 2514952\r\n",
      "   processing line 2521052\r\n",
      "   processing line 2527112\r\n",
      "   processing line 2533112\r\n",
      "   processing line 2539079\r\n",
      "   processing line 2544979\r\n",
      "   processing line 2550871\r\n",
      "   processing line 2556671\r\n",
      "   processing line 2562471\r\n",
      "   processing line 2568203\r\n",
      "   processing line 2573903\r\n",
      "   processing line 2579582\r\n",
      "   processing line 2585182\r\n",
      "   processing line 2590782\r\n",
      "   processing line 2596283\r\n",
      "   processing line 2601783\r\n",
      "   processing line 2607248\r\n",
      "   processing line 2612648\r\n",
      "   processing line 2618048\r\n",
      "   processing line 2623390\r\n",
      "   processing line 2628690\r\n",
      "   processing line 2633990\r\n",
      "   processing line 2639201\r\n",
      "   processing line 2644401\r\n",
      "   processing line 2649601\r\n",
      "   processing line 2654729\r\n",
      "   processing line 2659829\r\n",
      "   processing line 2664929\r\n",
      "   processing line 2669994\r\n",
      "   processing line 2674994\r\n",
      "   processing line 2679994\r\n",
      "   processing line 2684994\r\n",
      "   processing line 2689914\r\n",
      "   processing line 2694814\r\n",
      "   processing line 2699714\r\n",
      "   processing line 2704580\r\n",
      "   processing line 2709380\r\n",
      "   processing line 2714180\r\n",
      "   processing line 2718980\r\n",
      "   processing line 2723692\r\n",
      "   processing line 2728392\r\n",
      "   processing line 2733092\r\n",
      "   processing line 2737769\r\n",
      "   processing line 2742369\r\n",
      "   processing line 2746969\r\n",
      "   processing line 2751569\r\n",
      "   processing line 2756160\r\n",
      "   processing line 2760660\r\n",
      "   processing line 2765160\r\n",
      "   processing line 2769660\r\n",
      "   processing line 2774160\r\n",
      "   processing line 2778582\r\n",
      "   processing line 2782982\r\n",
      "   processing line 2787382\r\n",
      "   processing line 2791782\r\n",
      "   processing line 2796149\r\n",
      "   processing line 2800449\r\n",
      "   processing line 2804749\r\n",
      "   processing line 2809049\r\n",
      "   processing line 2813349\r\n",
      "   processing line 2817582\r\n",
      "   processing line 2821782\r\n",
      "   processing line 2825982\r\n",
      "   processing line 2830182\r\n",
      "   processing line 2834375\r\n",
      "   processing line 2838475\r\n",
      "   processing line 2842575\r\n",
      "   processing line 2846675\r\n",
      "   processing line 2850775\r\n",
      "   processing line 2854875\r\n",
      "   processing line 2858875\r\n",
      "   processing line 2862875\r\n",
      "   processing line 2866875\r\n",
      "   processing line 2870875\r\n",
      "   processing line 2874875\r\n",
      "   processing line 2878839\r\n",
      "   processing line 2882739\r\n",
      "   processing line 2886639\r\n",
      "   processing line 2890539\r\n",
      "   processing line 2894439\r\n",
      "   processing line 2898339\r\n",
      "   processing line 2902212\r\n",
      "   processing line 2906012\r\n",
      "   processing line 2909812\r\n",
      "   processing line 2913612\r\n",
      "   processing line 2917412\r\n",
      "   processing line 2921212\r\n",
      "   processing line 2924962\r\n",
      "   processing line 2928662\r\n",
      "   processing line 2932362\r\n",
      "   processing line 2936062\r\n",
      "   processing line 2939762\r\n",
      "   processing line 2943462\r\n",
      "   processing line 2947155\r\n",
      "   processing line 2950755\r\n",
      "   processing line 2954355\r\n",
      "   processing line 2957955\r\n",
      "   processing line 2961555\r\n",
      "   processing line 2965155\r\n",
      "   processing line 2968755\r\n",
      "   processing line 2972355\r\n",
      "   processing line 2975855\r\n",
      "   processing line 2979355\r\n",
      "   processing line 2982855\r\n",
      "   processing line 2986355\r\n",
      "   processing line 2989855\r\n",
      "   processing line 2993355\r\n",
      "   processing line 2996855\r\n",
      "   processing line 3000284\r\n",
      "   processing line 3003684\r\n",
      "   processing line 3007084\r\n",
      "   processing line 3010484\r\n",
      "   processing line 3013884\r\n",
      "   processing line 3017284\r\n",
      "   processing line 3020684\r\n",
      "   processing line 3024084\r\n",
      "   processing line 3027418\r\n",
      "   processing line 3030718\r\n",
      "   processing line 3034018\r\n",
      "   processing line 3037318\r\n",
      "   processing line 3040618\r\n",
      "   processing line 3043918\r\n",
      "   processing line 3047218\r\n",
      "   processing line 3050518\r\n",
      "   processing line 3053818\r\n",
      "   processing line 3057029\r\n",
      "   processing line 3060229\r\n",
      "   processing line 3063429\r\n",
      "   processing line 3066629\r\n",
      "   processing line 3069829\r\n",
      "   processing line 3073029\r\n",
      "   processing line 3076229\r\n",
      "   processing line 3079429\r\n",
      "   processing line 3082629\r\n",
      "   processing line 3085740\r\n",
      "   processing line 3088840\r\n",
      "   processing line 3091940\r\n",
      "   processing line 3095040\r\n",
      "   processing line 3098140\r\n",
      "   processing line 3101240\r\n",
      "   processing line 3104340\r\n",
      "   processing line 3107440\r\n",
      "   processing line 3110540\r\n",
      "   processing line 3113640\r\n",
      "   processing line 3116640\r\n",
      "   processing line 3119640\r\n",
      "   processing line 3122640\r\n",
      "   processing line 3125640\r\n",
      "   processing line 3128640\r\n",
      "   processing line 3131640\r\n",
      "   processing line 3134640\r\n",
      "   processing line 3137640\r\n",
      "   processing line 3140640\r\n",
      "   processing line 3143640\r\n",
      "   processing line 3146628\r\n",
      "   processing line 3149528\r\n",
      "   processing line 3152428\r\n",
      "   processing line 3155328\r\n",
      "   processing line 3158228\r\n",
      "   processing line 3161128\r\n",
      "   processing line 3164028\r\n",
      "   processing line 3166928\r\n",
      "   processing line 3169828\r\n",
      "   processing line 3172728\r\n",
      "   processing line 3175628\r\n",
      "   processing line 3178528\r\n",
      "   processing line 3181372\r\n",
      "   processing line 3184172\r\n",
      "   processing line 3186972\r\n",
      "   processing line 3189772\r\n",
      "   processing line 3192572\r\n",
      "   processing line 3195372\r\n",
      "   processing line 3198172\r\n",
      "   processing line 3200972\r\n",
      "   processing line 3203772\r\n",
      "   processing line 3206572\r\n",
      "   processing line 3209372\r\n",
      "   processing line 3212172\r\n",
      "   processing line 3214972\r\n",
      "   processing line 3217707\r\n",
      "   processing line 3220407\r\n",
      "   processing line 3223107\r\n",
      "   processing line 3225807\r\n",
      "   processing line 3228507\r\n",
      "   processing line 3231207\r\n",
      "   processing line 3233907\r\n",
      "   processing line 3236607\r\n",
      "   processing line 3239307\r\n",
      "   processing line 3242007\r\n",
      "   processing line 3244707\r\n",
      "   processing line 3247407\r\n",
      "   processing line 3250107\r\n",
      "   processing line 6584116\r\n",
      "3. Done. Thank you for using DeepVirFinder.\r\n",
      "   output in /xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_dvf/ERR2198704/contigs.fasta_gt1500bp_dvfpred.txt\r\n",
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses\r\n",
      "i16n3.ocelote.hpc.arizona.edu\r\n",
      "Sat Jul 20 20:54:06 MST 2024\r\n",
      "WARNING (theano.tensor.blas): Using NumPy C-API based implementation for BLAS functions.\r\n",
      "1. Loading Models.\r\n",
      "   model directory /groups/bhurwitz/databases/DeepVirFinder/models\r\n",
      "2. Encoding and Predicting Sequences.\r\n",
      "   processing line 1\r\n",
      "   processing line 366458\r\n",
      "Using Theano backend.\r\n",
      "   processing line 549752\r\n",
      "   processing line 689164\r\n",
      "   processing line 802098\r\n",
      "   processing line 897240\r\n",
      "   processing line 979284\r\n",
      "   processing line 1051615\r\n",
      "   processing line 1115131\r\n",
      "   processing line 1172827\r\n",
      "   processing line 1225377\r\n",
      "   processing line 1274197\r\n",
      "   processing line 1319358\r\n",
      "   processing line 1361311\r\n",
      "   processing line 1400413\r\n",
      "   processing line 1437127\r\n",
      "   processing line 1471428\r\n",
      "   processing line 1503970\r\n",
      "   processing line 1534772\r\n",
      "   processing line 1564029\r\n",
      "   processing line 1592131\r\n",
      "   processing line 1619123\r\n",
      "   processing line 1644890\r\n",
      "   processing line 1669359\r\n",
      "   processing line 1692942\r\n",
      "   processing line 1715683\r\n",
      "   processing line 1737651\r\n",
      "   processing line 1758898\r\n",
      "   processing line 1779321\r\n",
      "   processing line 1799044\r\n",
      "   processing line 1818189\r\n",
      "   processing line 1836747\r\n",
      "   processing line 1854800\r\n",
      "   processing line 1872373\r\n",
      "   processing line 1889471\r\n",
      "   processing line 1906094\r\n",
      "   processing line 1922223\r\n",
      "   processing line 1937901\r\n",
      "   processing line 1953162\r\n",
      "   processing line 1968061\r\n",
      "   processing line 1982594\r\n",
      "   processing line 1996760\r\n",
      "   processing line 2010611\r\n",
      "   processing line 2024186\r\n",
      "   processing line 2037467\r\n",
      "   processing line 2050472\r\n",
      "   processing line 2063237\r\n",
      "   processing line 2075735\r\n",
      "   processing line 2087957\r\n",
      "   processing line 2099943\r\n",
      "   processing line 2111714\r\n",
      "   processing line 2123279\r\n",
      "   processing line 2134639\r\n",
      "   processing line 2145811\r\n",
      "   processing line 2156790\r\n",
      "   processing line 2167600\r\n",
      "   processing line 2178197\r\n",
      "   processing line 2188637\r\n",
      "   processing line 2198906\r\n",
      "   processing line 2208986\r\n",
      "   processing line 2218927\r\n",
      "   processing line 2228726\r\n",
      "   processing line 2238382\r\n",
      "   processing line 2247906\r\n",
      "   processing line 2257302\r\n",
      "   processing line 2266568\r\n",
      "   processing line 2275726\r\n",
      "   processing line 2284766\r\n",
      "   processing line 2293690\r\n",
      "   processing line 2302504\r\n",
      "   processing line 2311190\r\n",
      "   processing line 2319760\r\n",
      "   processing line 2328231\r\n",
      "   processing line 2336585\r\n",
      "   processing line 2344834\r\n",
      "   processing line 2352983\r\n",
      "   processing line 2361061\r\n",
      "   processing line 2369038\r\n",
      "   processing line 2376916\r\n",
      "   processing line 2384705\r\n",
      "   processing line 2392391\r\n",
      "   processing line 2399991\r\n",
      "   processing line 2407492\r\n",
      "   processing line 2414914\r\n",
      "   processing line 2422267\r\n",
      "   processing line 2429541\r\n",
      "   processing line 2436741\r\n",
      "   processing line 2443860\r\n",
      "   processing line 2450928\r\n",
      "   processing line 2457928\r\n",
      "   processing line 2464834\r\n",
      "   processing line 2471691\r\n",
      "   processing line 2478488\r\n",
      "   processing line 2485188\r\n",
      "   processing line 2491844\r\n",
      "   processing line 2498444\r\n",
      "   processing line 2504953\r\n",
      "   processing line 2511427\r\n",
      "   processing line 2517827\r\n",
      "   processing line 2524150\r\n",
      "   processing line 2530450\r\n",
      "   processing line 2536661\r\n",
      "   processing line 2542851\r\n",
      "   processing line 2548951\r\n",
      "   processing line 2555018\r\n",
      "   processing line 2561018\r\n",
      "   processing line 2566993\r\n",
      "   processing line 2572893\r\n",
      "   processing line 2578746\r\n",
      "   processing line 2584546\r\n",
      "   processing line 2590294\r\n",
      "   processing line 2595994\r\n",
      "   processing line 2601624\r\n",
      "   processing line 2607224\r\n",
      "   processing line 2612800\r\n",
      "   processing line 2618300\r\n",
      "   processing line 2623800\r\n",
      "   processing line 2629204\r\n",
      "   processing line 2634604\r\n",
      "   processing line 2639968\r\n",
      "   processing line 2645268\r\n",
      "   processing line 2650568\r\n",
      "   processing line 2655809\r\n",
      "   processing line 2661009\r\n",
      "   processing line 2666209\r\n",
      "   processing line 2671360\r\n",
      "   processing line 2676460\r\n",
      "   processing line 2681560\r\n",
      "   processing line 2686619\r\n",
      "   processing line 2691619\r\n",
      "   processing line 2696619\r\n",
      "   processing line 2701583\r\n",
      "   processing line 2706483\r\n",
      "   processing line 2711383\r\n",
      "   processing line 2716283\r\n",
      "   processing line 2721095\r\n",
      "   processing line 2725895\r\n",
      "   processing line 2730695\r\n",
      "   processing line 2735449\r\n",
      "   processing line 2740149\r\n",
      "   processing line 2744849\r\n",
      "   processing line 2749549\r\n",
      "   processing line 2754170\r\n",
      "   processing line 2758770\r\n",
      "   processing line 2763370\r\n",
      "   processing line 2767951\r\n",
      "   processing line 2772451\r\n",
      "   processing line 2776951\r\n",
      "   processing line 2781451\r\n",
      "   processing line 2785935\r\n",
      "   processing line 2790335\r\n",
      "   processing line 2794735\r\n",
      "   processing line 2799135\r\n",
      "   processing line 2803535\r\n",
      "   processing line 2807836\r\n",
      "   processing line 2812136\r\n",
      "   processing line 2816436\r\n",
      "   processing line 2820736\r\n",
      "   processing line 2824955\r\n",
      "   processing line 2829155\r\n",
      "   processing line 2833355\r\n",
      "   processing line 2837555\r\n",
      "   processing line 2841744\r\n",
      "   processing line 2845844\r\n",
      "   processing line 2849944\r\n",
      "   processing line 2854044\r\n",
      "   processing line 2858144\r\n",
      "   processing line 2862244\r\n",
      "   processing line 2866257\r\n",
      "   processing line 2870257\r\n",
      "   processing line 2874257\r\n",
      "   processing line 2878257\r\n",
      "   processing line 2882257\r\n",
      "   processing line 2886214\r\n",
      "   processing line 2890114\r\n",
      "   processing line 2894014\r\n",
      "   processing line 2897914\r\n",
      "   processing line 2901814\r\n",
      "   processing line 2905714\r\n",
      "   processing line 2909544\r\n",
      "   processing line 2913344\r\n",
      "   processing line 2917144\r\n",
      "   processing line 2920944\r\n",
      "   processing line 2924744\r\n",
      "   processing line 2928523\r\n",
      "   processing line 2932223\r\n",
      "   processing line 2935923\r\n",
      "   processing line 2939623\r\n",
      "   processing line 2943323\r\n",
      "   processing line 2947023\r\n",
      "   processing line 2950723\r\n",
      "   processing line 2954342\r\n",
      "   processing line 2957942\r\n",
      "   processing line 2961542\r\n",
      "   processing line 2965142\r\n",
      "   processing line 2968742\r\n",
      "   processing line 2972342\r\n",
      "   processing line 2975924\r\n",
      "   processing line 2979424\r\n",
      "   processing line 2982924\r\n",
      "   processing line 2986424\r\n",
      "   processing line 2989924\r\n",
      "   processing line 2993424\r\n",
      "   processing line 2996924\r\n",
      "   processing line 3000424\r\n",
      "   processing line 3003835\r\n",
      "   processing line 3007235\r\n",
      "   processing line 3010635\r\n",
      "   processing line 3014035\r\n",
      "   processing line 3017435\r\n",
      "   processing line 3020835\r\n",
      "   processing line 3024235\r\n",
      "   processing line 3027608\r\n",
      "   processing line 3030908\r\n",
      "   processing line 3034208\r\n",
      "   processing line 3037508\r\n",
      "   processing line 3040808\r\n",
      "   processing line 3044108\r\n",
      "   processing line 3047408\r\n",
      "   processing line 3050708\r\n",
      "   processing line 3054008\r\n",
      "   processing line 3057221\r\n",
      "   processing line 3060421\r\n",
      "   processing line 3063621\r\n",
      "   processing line 3066821\r\n",
      "   processing line 3070021\r\n",
      "   processing line 3073221\r\n",
      "   processing line 3076421\r\n",
      "   processing line 3079621\r\n",
      "   processing line 3082821\r\n",
      "   processing line 3085939\r\n",
      "   processing line 3089039\r\n",
      "   processing line 3092139\r\n",
      "   processing line 3095239\r\n",
      "   processing line 3098339\r\n",
      "   processing line 3101439\r\n",
      "   processing line 3104539\r\n",
      "   processing line 3107639\r\n",
      "   processing line 3110739\r\n",
      "   processing line 3113839\r\n",
      "   processing line 3116853\r\n",
      "   processing line 3119853\r\n",
      "   processing line 3122853\r\n",
      "   processing line 3125853\r\n",
      "   processing line 3128853\r\n",
      "   processing line 3131853\r\n",
      "   processing line 3134853\r\n",
      "   processing line 3137853\r\n",
      "   processing line 3140853\r\n",
      "   processing line 3143853\r\n",
      "   processing line 3146796\r\n",
      "   processing line 3149696\r\n",
      "   processing line 3152596\r\n",
      "   processing line 3155496\r\n",
      "   processing line 3158396\r\n",
      "   processing line 3161296\r\n",
      "   processing line 3164196\r\n",
      "   processing line 3167096\r\n",
      "   processing line 3169996\r\n",
      "   processing line 3172896\r\n",
      "   processing line 3175796\r\n",
      "   processing line 3178679\r\n",
      "   processing line 3181479\r\n",
      "   processing line 3184279\r\n",
      "   processing line 3187079\r\n",
      "   processing line 3189879\r\n",
      "   processing line 3192679\r\n",
      "   processing line 3195479\r\n",
      "   processing line 3198279\r\n",
      "   processing line 3201079\r\n",
      "   processing line 3203879\r\n",
      "   processing line 3206679\r\n",
      "   processing line 3209479\r\n",
      "   processing line 3212279\r\n",
      "   processing line 3214981\r\n",
      "   processing line 3217681\r\n",
      "   processing line 3220381\r\n",
      "   processing line 3223081\r\n",
      "   processing line 3225781\r\n",
      "   processing line 3228481\r\n",
      "   processing line 3231181\r\n",
      "   processing line 3233881\r\n",
      "   processing line 3236581\r\n",
      "   processing line 3239281\r\n",
      "   processing line 3241981\r\n",
      "   processing line 3244681\r\n",
      "   processing line 3247381\r\n",
      "   processing line 6568866\r\n",
      "3. Done. Thank you for using DeepVirFinder.\r\n",
      "   output in /xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_dvf/ERR2198705/contigs.fasta_gt1500bp_dvfpred.txt\r\n"
     ]
    }
   ],
   "source": [
    "!cat Job-dvf-*.out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "e636764d",
   "metadata": {},
   "outputs": [],
   "source": [
    "my_code = '''#!/bin/bash\n",
    "#SBATCH --output=Job-checkv-dvf-%a.out\n",
    "#SBATCH --account=bhurwitz\n",
    "#SBATCH --partition=standard\n",
    "#SBATCH --nodes=1\n",
    "#SBATCH --ntasks=1\n",
    "#SBATCH --time=20:00:00\n",
    "#SBATCH --cpus-per-task=4\n",
    "#SBATCH --mem-per-cpu=1000\n",
    "#SBATCH --array=0-2\n",
    "\n",
    "pwd; hostname; date\n",
    "\n",
    "source ./config.sh\n",
    "names=($(cat $XFILE_DIR/$XFILE))\n",
    "SAMPLE_ID=${names[${SLURM_ARRAY_TASK_ID}]}\n",
    "\n",
    "DVF=${OUT_DVF}/${SAMPLE_ID}/contigs.fasta_gt1500bp_dvfpred.txt\n",
    "\n",
    "SPADES=${SPADES_DIR}/${SAMPLE_ID}/contigs.fasta\n",
    "\n",
    "DVF_FASTA=${OUT_DVF}/${SAMPLE_ID}/dvf.fasta\n",
    "\n",
    "#load environment\n",
    "CONDA=\"/groups/bhurwitz/miniconda3\"\n",
    "source $CONDA/etc/profile.d/conda.sh\n",
    "conda activate checkv_env\n",
    "\n",
    "./id_from_fasta.py -c ${SPADES} -d ${DVF} -o ${DVF_FASTA}\n",
    "\n",
    "checkv end_to_end ${DVF_FASTA} ${OUT_CHECKV_DVF}/${names[${SLURM_ARRAY_TASK_ID}]} -t 16\n",
    "'''\n",
    "\n",
    "with open('run_checkv_dvf.sh', mode='w') as file:\n",
    "    file.write(my_code)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "6b5b664a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses\n",
      "DeepVirFinder\t\t   Job-dvf-2.out       out_dvf\n",
      "Job-checkv-dvf-0.out\t   Job-virsort2-0.out  out_virsort\n",
      "Job-checkv-dvf-1.out\t   Job-virsort2-1.out  run_checkv_dvf.sh\n",
      "Job-checkv-dvf-2.out\t   Job-virsort2-2.out  run_checkv_virsort2.sh\n",
      "Job-checkv-virsort2-0.out  config.sh\t       run_dvf.sh\n",
      "Job-checkv-virsort2-1.out  dvf.fasta\t       run_virsort2.sh\n",
      "Job-checkv-virsort2-2.out  id_from_fasta.py    temp.sh\n",
      "Job-dvf-0.out\t\t   out_checkv_dvf\n",
      "Job-dvf-1.out\t\t   out_checkv_virsort\n"
     ]
    }
   ],
   "source": [
    "#check that I am in working directory with config file and run_virsort2.sh script\n",
    "!pwd\n",
    "!ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "95006688",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Submitted batch job 3291969\r\n"
     ]
    }
   ],
   "source": [
    "#run script!\n",
    "!sbatch ./run_checkv_dvf.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "446ec26c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n",
      "         3291969_0  standard run_chec kolodisn  R       1:49      1 i1n3\n",
      "         3291969_1  standard run_chec kolodisn  R       1:49      1 i6n0\n",
      "         3291969_2  standard run_chec kolodisn  R       1:49      1 i6n0\n",
      "           3291967  standard  jupyter kolodisn  R       6:49      1 i16n10\n"
     ]
    }
   ],
   "source": [
    "#is the job in queue? \n",
    "!squeue --user=$netid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "ece32c18",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_checkv_dvf/ERR2198703:\r\n",
      "complete_genomes.tsv  contamination.tsv  quality_summary.tsv  viruses.fna\r\n",
      "completeness.tsv      proviruses.fna\t tmp\r\n",
      "\r\n",
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_checkv_dvf/ERR2198704:\r\n",
      "complete_genomes.tsv  contamination.tsv  quality_summary.tsv  viruses.fna\r\n",
      "completeness.tsv      proviruses.fna\t tmp\r\n",
      "\r\n",
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_checkv_dvf/ERR2198705:\r\n",
      "complete_genomes.tsv  contamination.tsv  quality_summary.tsv  viruses.fna\r\n",
      "completeness.tsv      proviruses.fna\t tmp\r\n"
     ]
    }
   ],
   "source": [
    "#check the output files \n",
    "!ls $work_dir/out_checkv_dvf/*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "2a3c1d19",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses\r\n",
      "i1n3.ocelote.hpc.arizona.edu\r\n",
      "Mon Jul  8 12:02:10 MST 2024\r\n",
      "\r\n",
      "CheckV v0.9.0: contamination\r\n",
      "[1/8] Reading database info...\r\n",
      "[2/8] Reading genome info...\r\n",
      "[3/8] Skipping gene calling...\r\n",
      "[4/8] Reading gene info...\r\n",
      "[5/8] Skipping hmmsearch...\r\n",
      "[6/8] Annotating genes...\r\n",
      "[7/8] Identifying host regions...\r\n",
      "[8/8] Writing results...\r\n",
      "Run time: 18.4 seconds\r\n",
      "Peak mem: 0.33 GB\r\n",
      "\r\n",
      "CheckV v0.9.0: completeness\r\n",
      "[1/8] Skipping gene calling...\r\n",
      "[2/8] Initializing queries and database...\r\n",
      "[3/8] Running DIAMOND blastp search...\r\n",
      "[4/8] Computing AAI...\r\n",
      "[5/8] Running AAI based completeness estimation...\r\n",
      "[6/8] Running HMM based completeness estimation...\r\n",
      "[7/8] Determining genome copy number...\r\n",
      "[8/8] Writing results...\r\n",
      "Run time: 176.79 seconds\r\n",
      "Peak mem: 3.76 GB\r\n",
      "\r\n",
      "CheckV v0.9.0: complete_genomes\r\n",
      "[1/7] Reading input sequences...\r\n",
      "[2/7] Finding complete proviruses...\r\n",
      "[3/7] Finding direct/inverted terminal repeats...\r\n",
      "[4/7] Filtering terminal repeats...\r\n",
      "[5/7] Checking genome for completeness...\r\n",
      "[6/7] Checking genome for large duplications...\r\n",
      "[7/7] Writing results...\r\n",
      "Run time: 1.98 seconds\r\n",
      "Peak mem: 3.76 GB\r\n",
      "\r\n",
      "CheckV v0.9.0: quality_summary\r\n",
      "[1/6] Reading input sequences...\r\n",
      "[2/6] Reading results from contamination module...\r\n",
      "[3/6] Reading results from completeness module...\r\n",
      "[4/6] Reading results from complete genomes module...\r\n",
      "[5/6] Classifying contigs into quality tiers...\r\n",
      "[6/6] Writing results...\r\n",
      "Run time: 1.17 seconds\r\n",
      "Peak mem: 3.76 GB\r\n",
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses\r\n",
      "i6n0.ocelote.hpc.arizona.edu\r\n",
      "Mon Jul  8 12:02:10 MST 2024\r\n",
      "\r\n",
      "CheckV v0.9.0: contamination\r\n",
      "[1/8] Reading database info...\r\n",
      "[2/8] Reading genome info...\r\n",
      "[3/8] Skipping gene calling...\r\n",
      "[4/8] Reading gene info...\r\n",
      "[5/8] Skipping hmmsearch...\r\n",
      "[6/8] Annotating genes...\r\n",
      "[7/8] Identifying host regions...\r\n",
      "[8/8] Writing results...\r\n",
      "Run time: 27.97 seconds\r\n",
      "Peak mem: 0.46 GB\r\n",
      "\r\n",
      "CheckV v0.9.0: completeness\r\n",
      "[1/8] Skipping gene calling...\r\n",
      "[2/8] Initializing queries and database...\r\n",
      "[3/8] Running DIAMOND blastp search...\r\n",
      "[4/8] Computing AAI...\r\n",
      "[5/8] Running AAI based completeness estimation...\r\n",
      "[6/8] Running HMM based completeness estimation...\r\n",
      "[7/8] Determining genome copy number...\r\n",
      "[8/8] Writing results...\r\n",
      "Run time: 309.72 seconds\r\n",
      "Peak mem: 4.21 GB\r\n",
      "\r\n",
      "CheckV v0.9.0: complete_genomes\r\n",
      "[1/7] Reading input sequences...\r\n",
      "[2/7] Finding complete proviruses...\r\n",
      "[3/7] Finding direct/inverted terminal repeats...\r\n",
      "[4/7] Filtering terminal repeats...\r\n",
      "[5/7] Checking genome for completeness...\r\n",
      "[6/7] Checking genome for large duplications...\r\n",
      "[7/7] Writing results...\r\n",
      "Run time: 3.13 seconds\r\n",
      "Peak mem: 4.21 GB\r\n",
      "\r\n",
      "CheckV v0.9.0: quality_summary\r\n",
      "[1/6] Reading input sequences...\r\n",
      "[2/6] Reading results from contamination module...\r\n",
      "[3/6] Reading results from completeness module...\r\n",
      "[4/6] Reading results from complete genomes module...\r\n",
      "[5/6] Classifying contigs into quality tiers...\r\n",
      "[6/6] Writing results...\r\n",
      "Run time: 1.92 seconds\r\n",
      "Peak mem: 4.21 GB\r\n",
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses\r\n",
      "i6n0.ocelote.hpc.arizona.edu\r\n",
      "Mon Jul  8 12:02:10 MST 2024\r\n",
      "\r\n",
      "CheckV v0.9.0: contamination\r\n",
      "[1/8] Reading database info...\r\n",
      "[2/8] Reading genome info...\r\n",
      "[3/8] Skipping gene calling...\r\n",
      "[4/8] Reading gene info...\r\n",
      "[5/8] Skipping hmmsearch...\r\n",
      "[6/8] Annotating genes...\r\n",
      "[7/8] Identifying host regions...\r\n",
      "[8/8] Writing results...\r\n",
      "Run time: 28.78 seconds\r\n",
      "Peak mem: 0.45 GB\r\n",
      "\r\n",
      "CheckV v0.9.0: completeness\r\n",
      "[1/8] Skipping gene calling...\r\n",
      "[2/8] Initializing queries and database...\r\n",
      "[3/8] Running DIAMOND blastp search...\r\n",
      "[4/8] Computing AAI...\r\n",
      "[5/8] Running AAI based completeness estimation...\r\n",
      "[6/8] Running HMM based completeness estimation...\r\n",
      "[7/8] Determining genome copy number...\r\n",
      "[8/8] Writing results...\r\n",
      "Run time: 320.63 seconds\r\n",
      "Peak mem: 4.18 GB\r\n",
      "\r\n",
      "CheckV v0.9.0: complete_genomes\r\n",
      "[1/7] Reading input sequences...\r\n",
      "[2/7] Finding complete proviruses...\r\n",
      "[3/7] Finding direct/inverted terminal repeats...\r\n",
      "[4/7] Filtering terminal repeats...\r\n",
      "[5/7] Checking genome for completeness...\r\n",
      "[6/7] Checking genome for large duplications...\r\n",
      "[7/7] Writing results...\r\n",
      "Run time: 3.15 seconds\r\n",
      "Peak mem: 4.18 GB\r\n",
      "\r\n",
      "CheckV v0.9.0: quality_summary\r\n",
      "[1/6] Reading input sequences...\r\n",
      "[2/6] Reading results from contamination module...\r\n",
      "[3/6] Reading results from completeness module...\r\n",
      "[4/6] Reading results from complete genomes module...\r\n",
      "[5/6] Classifying contigs into quality tiers...\r\n",
      "[6/6] Writing results...\r\n",
      "Run time: 1.95 seconds\r\n",
      "Peak mem: 4.18 GB\r\n"
     ]
    }
   ],
   "source": [
    "!cat Job-checkv-dvf-*.out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "9b2e0e81",
   "metadata": {},
   "outputs": [],
   "source": [
    "my_code = '''#!/bin/bash\n",
    "#SBATCH --output=Job-genomad-%a.out\n",
    "#SBATCH --account=bhurwitz\n",
    "#SBATCH --partition=standard\n",
    "#SBATCH --nodes=1\n",
    "#SBATCH --ntasks=1\n",
    "#SBATCH --time=20:00:00\n",
    "#SBATCH --cpus-per-task=4\n",
    "#SBATCH --mem-per-cpu=1000\n",
    "#SBATCH --array=0-2\n",
    "\n",
    "pwd; hostname; date\n",
    "\n",
    "source ./config.sh\n",
    "names=($(cat $XFILE_DIR/$XFILE))\n",
    "SAMPLE_ID=${names[${SLURM_ARRAY_TASK_ID}]}\n",
    "\n",
    "SPADES=${SPADES_DIR}/${SAMPLE_ID}/contigs.fasta.gz\n",
    "\n",
    "#load environment\n",
    "CONDA=\"/groups/bhurwitz/miniconda3\"\n",
    "source $CONDA/etc/profile.d/conda.sh\n",
    "conda activate genomad_env\n",
    "\n",
    "genomad end-to-end --cleanup ${SPADES} ${OUT_GENOMAD}/${names[${SLURM_ARRAY_TASK_ID}]} ${GENOMAD_DB}\n",
    "'''\n",
    "\n",
    "with open('run_genomad.sh', mode='w') as file:\n",
    "    file.write(my_code)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "bef67a8c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses\n",
      "06_get_viruses.ipynb\t   Job-dvf-2.out       out_checkv_virsort\n",
      "Job-checkv-dvf-0.out\t   Job-genomad-0.out   out_dvf\n",
      "Job-checkv-dvf-1.out\t   Job-genomad-1.out   out_genomad\n",
      "Job-checkv-dvf-2.out\t   Job-genomad-2.out   out_virsort\n",
      "Job-checkv-genomad-0.out   Job-virsort2-0.out  run_checkv_dvf.sh\n",
      "Job-checkv-genomad-1.out   Job-virsort2-1.out  run_checkv_genomad.sh\n",
      "Job-checkv-genomad-2.out   Job-virsort2-2.out  run_checkv_virsort2.sh\n",
      "Job-checkv-virsort2-0.out  config.sh\t       run_dvf.sh\n",
      "Job-checkv-virsort2-1.out  dvf.fasta\t       run_genomad.sh\n",
      "Job-checkv-virsort2-2.out  id_from_fasta.py    run_virsort2.sh\n",
      "Job-dvf-0.out\t\t   out_checkv_dvf      temp.sh\n",
      "Job-dvf-1.out\t\t   out_checkv_genomad\n"
     ]
    }
   ],
   "source": [
    "#check that I am in working directory with config file and run_virsort2.sh script\n",
    "!pwd\n",
    "!ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "f5707fec",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Submitted batch job 3296365\r\n"
     ]
    }
   ],
   "source": [
    "#run script!\n",
    "!sbatch ./run_genomad.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "8ca04a0c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n",
      "           3296375  standard  jupyter kolodisn PD       0:00      1 (Priority)\n",
      "         3296365_1  standard run_geno kolodisn  R      13:45      1 i16n1\n",
      "         3296365_2  standard run_geno kolodisn  R      13:45      1 i16n1\n",
      "           3296354  standard  jupyter kolodisn  R      58:46      1 i16n1\n"
     ]
    }
   ],
   "source": [
    "#is the job in queue? \n",
    "!squeue --user=$netid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "db691b81",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_genomad/ERR2198703:\r\n",
      "contigs_aggregated_classification      contigs_marker_classification\r\n",
      "contigs_aggregated_classification.log  contigs_marker_classification.log\r\n",
      "contigs_annotate\t\t       contigs_nn_classification\r\n",
      "contigs_annotate.log\t\t       contigs_nn_classification.log\r\n",
      "contigs_find_proviruses\t\t       contigs_summary\r\n",
      "contigs_find_proviruses.log\t       contigs_summary.log\r\n",
      "\r\n",
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_genomad/ERR2198704:\r\n",
      "contigs_aggregated_classification      contigs_marker_classification\r\n",
      "contigs_aggregated_classification.log  contigs_marker_classification.log\r\n",
      "contigs_annotate\t\t       contigs_nn_classification\r\n",
      "contigs_annotate.log\t\t       contigs_nn_classification.log\r\n",
      "contigs_find_proviruses\t\t       contigs_summary\r\n",
      "contigs_find_proviruses.log\t       contigs_summary.log\r\n",
      "\r\n",
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_genomad/ERR2198705:\r\n",
      "contigs_aggregated_classification      contigs_marker_classification\r\n",
      "contigs_aggregated_classification.log  contigs_marker_classification.log\r\n",
      "contigs_annotate\t\t       contigs_nn_classification\r\n",
      "contigs_annotate.log\t\t       contigs_nn_classification.log\r\n",
      "contigs_find_proviruses\t\t       contigs_summary\r\n",
      "contigs_find_proviruses.log\t       contigs_summary.log\r\n"
     ]
    }
   ],
   "source": [
    "#check the output files \n",
    "!ls $work_dir/out_genomad/*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "297b5484",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses\r\n",
      "i16n2.ocelote.hpc.arizona.edu\r\n",
      "Sat Jul 20 20:44:05 MST 2024\r\n",
      "\r\n",
      "  Executing geNomad annotate (v1.8.0). This will perform gene calling in the  \r\n",
      "  input sequences and annotate the predicted proteins with geNomad's          \r\n",
      "  markers.                                                                    \r\n",
      "    \r\n",
      "  Outputs:                                                                    \r\n",
      "    /xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_genomad/ERR2  \r\n",
      "    198704/contigs_annotate                                                   \r\n",
      "     contigs_annotate.json (execution parameters)                          \r\n",
      "     contigs_genes.tsv (gene annotation data)                              \r\n",
      "     contigs_taxonomy.tsv (taxonomic assignment)                           \r\n",
      "     contigs_mmseqs2.tsv (MMseqs2 output file)                             \r\n",
      "     contigs_proteins.faa (protein FASTA file)                             \r\n",
      "\r\n",
      "[20:44:12] Executing genomad annotate.                                          \r\n",
      "[20:44:13] Previous execution detected. Steps will be skipped unless their      \r\n",
      "           outputs are not found. Use the --restart option to force the         \r\n",
      "           execution of all the steps again.                                    \r\n",
      "[20:44:14] contigs_proteins.faa was found. Skipping gene prediction with        \r\n",
      "           pyrodigal-gv.                                                        \r\n",
      "[20:44:14] contigs_mmseqs2.tsv was found. Skipping protein annotation with      \r\n",
      "           MMseqs2.                                                             \r\n",
      "[20:44:24] Gene data was written to contigs_genes.tsv.                          \r\n",
      "[20:44:25] Taxonomic assignment data was written to contigs_taxonomy.tsv.       \r\n",
      "[20:44:25] geNomad annotate finished!                                           \r\n",
      "\r\n",
      "  Executing geNomad find-proviruses (v1.8.0). This will find putative         \r\n",
      "  proviral regions within the input sequences.                                \r\n",
      "    \r\n",
      "  Outputs:                                                                    \r\n",
      "    /xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_genomad/ERR2  \r\n",
      "    198704/contigs_find_proviruses                                            \r\n",
      "     contigs_find_proviruses.json (execution parameters)                   \r\n",
      "     contigs_provirus.tsv (provirus data)                                  \r\n",
      "     contigs_provirus.fna (provirus nucleotide sequences)                  \r\n",
      "     contigs_provirus_proteins.faa (provirus protein sequences)            \r\n",
      "     contigs_provirus_genes.tsv (provirus gene annotation data)            \r\n",
      "     contigs_provirus_taxonomy.tsv (provirus taxonomic assignment)         \r\n",
      "     contigs_provirus_mmseqs2.tsv (MMseqs2 output file)                    \r\n",
      "     contigs_provirus_aragorn.tsv (Aragorn output file)                    \r\n",
      "\r\n",
      "[20:44:28] Executing genomad find-proviruses.                                   \r\n",
      "[20:44:29] Previous execution detected. Steps will be skipped unless their      \r\n",
      "           outputs are not found. Use the --restart option to force the         \r\n",
      "           execution of all the steps again.                                    \r\n",
      "[20:44:34] contigs_provirus_mmseqs2.tsv was found. Skipping integrase search.   \r\n",
      "[20:44:34] contigs_provirus_aragorn.tsv was found. Skipping tRNA identification.\r\n",
      "[20:44:40] Provirus regions identified.                                         \r\n",
      "[20:44:40] Provirus data was written to contigs_provirus.tsv.                   \r\n",
      "[20:44:42] Provirus nucleotide sequences were written to contigs_provirus.fna.  \r\n",
      "[20:44:44] Provirus protein sequences were written to                           \r\n",
      "           contigs_provirus_proteins.faa.                                       \r\n",
      "[20:44:45] Provirus gene data was written to contigs_provirus_genes.tsv.        \r\n",
      "[20:44:45] Taxonomic assignment data was written to                             \r\n",
      "           contigs_provirus_taxonomy.tsv.                                       \r\n",
      "[20:44:45] geNomad find-proviruses finished!                                    \r\n",
      "\r\n",
      "  Executing geNomad marker-classification (v1.8.0). This will classify the    \r\n",
      "  input sequences into chromosome, plasmid, or virus based on the presence    \r\n",
      "  of geNomad markers and other gene-related features.                         \r\n",
      "    \r\n",
      "  Outputs:                                                                    \r\n",
      "    /xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_genomad/ERR2  \r\n",
      "    198704/contigs_marker_classification                                      \r\n",
      "     contigs_marker_classification.json (execution parameters)             \r\n",
      "     contigs_features.tsv (sequence feature data: tabular format)          \r\n",
      "     contigs_features.npz (sequence feature data: binary format)           \r\n",
      "     contigs_marker_classification.tsv (sequence classification: tabular   \r\n",
      "       format)                                                               \r\n",
      "     contigs_marker_classification.npz (sequence classification: binary    \r\n",
      "       format)                                                               \r\n",
      "     contigs_provirus_features.tsv (provirus feature data: tabular         \r\n",
      "       format)                                                               \r\n",
      "     contigs_provirus_features.npz (provirus feature data: binary format)  \r\n",
      "     contigs_provirus_marker_classification.tsv (provirus classification:  \r\n",
      "       tabular format)                                                       \r\n",
      "     contigs_provirus_marker_classification.npz (provirus classification:  \r\n",
      "        binary format)                                                        \r\n",
      "\r\n",
      "[20:44:49] Executing genomad marker-classification.                             \r\n",
      "[20:44:50] Previous execution detected. Steps will be skipped unless their      \r\n",
      "           outputs are not found. Use the --restart option to force the         \r\n",
      "           execution of all the steps again.                                    \r\n",
      "[20:44:50] contigs_features.npz was found. Skipping feature computation.        \r\n",
      "[20:45:00] Sequence features in tabular format written to contigs_features.tsv. \r\n",
      "[20:45:00] contigs_provirus_features.npz was found. Skipping provirus feature   \r\n",
      "           computation.                                                         \r\n",
      "[20:45:00] Provirus features in tabular format written to                       \r\n",
      "           contigs_provirus_features.tsv.                                       \r\n",
      "[20:45:00] contigs_marker_classification.npz was found. Skipping sequence       \r\n",
      "           classification.                                                      \r\n",
      "[20:45:02] Sequence classification in tabular format written to                 \r\n",
      "           contigs_marker_classification.tsv.                                   \r\n",
      "[20:45:02] contigs_provirus_marker_classification.npz was found. Skipping       \r\n",
      "           provirus classification.                                             \r\n",
      "[20:45:02] Provirus classification in tabular format written to                 \r\n",
      "           contigs_provirus_marker_classification.tsv.                          \r\n",
      "[20:45:02] geNomad marker-classification finished!                              \r\n",
      "\r\n",
      "  Executing geNomad nn-classification (v1.8.0). This will classify the input  \r\n",
      "  sequences into chromosome, plasmid, or virus based on the nucleotide        \r\n",
      "  sequence.                                                                   \r\n",
      "    \r\n",
      "  Outputs:                                                                    \r\n",
      "    /xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_genomad/ERR2  \r\n",
      "    198704/contigs_nn_classification                                          \r\n",
      "     contigs_nn_classification.json (execution parameters)                 \r\n",
      "     contigs_encoded_sequences (directory containing encoded sequence      \r\n",
      "       data)                                                                 \r\n",
      "     contigs_nn_classification.tsv (contig classification: tabular         \r\n",
      "       format)                                                               \r\n",
      "     contigs_nn_classification.npz (contig classification: binary format)  \r\n",
      "     contigs_encoded_proviruses (directory containing encoded sequence     \r\n",
      "       data)                                                                 \r\n",
      "     contigs_provirus_nn_classification.tsv (provirus classification:      \r\n",
      "       tabular format)                                                       \r\n",
      "     contigs_provirus_nn_classification.npz (provirus classification:      \r\n",
      "        binary format)                                                        \r\n",
      "\r\n",
      "[20:45:14] Executing genomad nn-classification.                                 \r\n",
      "[20:45:15] Previous execution detected. Steps will be skipped unless their      \r\n",
      "           outputs are not found. Use the --restart option to force the         \r\n",
      "           execution of all the steps again.                                    \r\n",
      "[20:45:15] Creating the                                                         \r\n",
      "           /xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_genomad/E\r\n",
      "           RR2198704/contigs_nn_classification/contigs_encoded_sequences        \r\n",
      "           directory.                                                           \r\n",
      "[20:59:18] Encoded sequence data written to contigs_encoded_sequences.          \r\n",
      "[20:59:18] Creating the                                                         \r\n",
      "           /xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_genomad/E\r\n",
      "           RR2198704/contigs_nn_classification/contigs_encoded_proviruses       \r\n",
      "           directory.                                                           \r\n",
      "[20:59:18] Encoded provirus data written to contigs_encoded_proviruses.         \r\n",
      "[20:59:18] contigs_nn_classification.npz was found. Skipping sequence           \r\n",
      "           classification.                                                      \r\n",
      "[20:59:18] Deleting encoded sequence data.                                      \r\n",
      "[20:59:21] Sequence classification in tabular format written to                 \r\n",
      "           contigs_nn_classification.tsv.                                       \r\n",
      "[20:59:21] contigs_provirus_nn_classification.npz was found. Skipping provirus  \r\n",
      "           classification.                                                      \r\n",
      "[20:59:21] Deleting encoded provirus data.                                      \r\n",
      "[20:59:21] Provirus classification in tabular format written to                 \r\n",
      "           contigs_provirus_nn_classification.tsv.                              \r\n",
      "[20:59:21] geNomad nn-classification finished!                                  \r\n",
      "\r\n",
      "  Executing geNomad aggregated-classification (v1.8.0). This will aggregate   \r\n",
      "  the results of the marker-classification and nn-classification modules to   \r\n",
      "  classify the input sequences into chromosome, plasmid, or virus.            \r\n",
      "    \r\n",
      "  Outputs:                                                                    \r\n",
      "    /xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_genomad/ERR2  \r\n",
      "    198704/contigs_aggregated_classification                                  \r\n",
      "     contigs_aggregated_classification.json (execution parameters)         \r\n",
      "     contigs_aggregated_classification.tsv (sequence classification:       \r\n",
      "       tabular format)                                                       \r\n",
      "     contigs_aggregated_classification.npz (sequence classification:       \r\n",
      "       binary format)                                                        \r\n",
      "     contigs_provirus_aggregated_classification.tsv (provirus              \r\n",
      "       classification: tabular format)                                       \r\n",
      "     contigs_provirus_aggregated_classification.npz (provirus              \r\n",
      "        classification: binary format)                                        \r\n",
      "\r\n",
      "[20:59:24] Executing genomad aggregated-classification.                         \r\n",
      "[20:59:25] Previous execution detected. Steps will be skipped unless their      \r\n",
      "           outputs are not found. Use the --restart option to force the         \r\n",
      "           execution of all the steps again.                                    \r\n",
      "[20:59:26] The total marker frequencies of the input sequences were computed.   \r\n",
      "[20:59:26] contigs_aggregated_classification.npz was found. Skipping sequence   \r\n",
      "           classification.                                                      \r\n",
      "[20:59:28] Sequence classification in tabular format written to                 \r\n",
      "           contigs_aggregated_classification.tsv.                               \r\n",
      "[20:59:28] contigs_provirus_aggregated_classification.npz was found. Skipping   \r\n",
      "           provirus classification.                                             \r\n",
      "[20:59:28] Provirus classification in tabular format written to                 \r\n",
      "           contigs_provirus_aggregated_classification.tsv.                      \r\n",
      "[20:59:28] geNomad aggregated-classification finished!                          \r\n",
      "\r\n",
      "  Executing geNomad summary (v1.8.0). This will summarize the results across  \r\n",
      "  modules into a classification report.                                       \r\n",
      "    \r\n",
      "  Outputs:                                                                    \r\n",
      "    /xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_genomad/ERR2  \r\n",
      "    198704/contigs_summary                                                    \r\n",
      "     contigs_summary.json (execution parameters)                           \r\n",
      "     contigs_virus_summary.tsv (virus classification summary)              \r\n",
      "     contigs_plasmid_summary.tsv (plasmid classification summary)          \r\n",
      "     contigs_virus.fna (virus nucleotide FASTA file)                       \r\n",
      "     contigs_plasmid.fna (plasmid nucleotide FASTA file)                   \r\n",
      "     contigs_virus_proteins.faa (virus protein FASTA file)                 \r\n",
      "     contigs_plasmid_proteins.faa (plasmid protein FASTA file)             \r\n",
      "     contigs_virus_genes.tsv (virus gene annotation data)                  \r\n",
      "     contigs_plasmid_genes.tsv (plasmid gene annotation data)              \r\n",
      "\r\n",
      "[20:59:29] Executing genomad summary.                                           \r\n",
      "[20:59:30] Using scores from aggregated-classification.                         \r\n",
      "[20:59:39] 852 plasmid(s) and 1,780 virus(es) were identified.                  \r\n",
      "[20:59:42] Nucleotide sequences were written to contigs_plasmid.fna and         \r\n",
      "           contigs_virus.fna.                                                   \r\n",
      "[20:59:45] Protein sequences were written to contigs_plasmid_proteins.faa and   \r\n",
      "           contigs_virus_proteins.faa.                                          \r\n",
      "[20:59:46] Gene annotation data was written to contigs_plasmid_genes.tsv and    \r\n",
      "           contigs_virus_genes.tsv.                                             \r\n",
      "[20:59:46] Summary files were written to contigs_plasmid_summary.tsv and        \r\n",
      "           contigs_virus_summary.tsv.                                           \r\n",
      "[20:59:46] geNomad summary finished!                                            \r\n"
     ]
    }
   ],
   "source": [
    "#check Job output!\n",
    "!cat Job-genomad-1.out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "7ebeed1d",
   "metadata": {},
   "outputs": [],
   "source": [
    "my_code = '''#!/bin/bash\n",
    "#SBATCH --output=Job-checkv-genomad-%a.out\n",
    "#SBATCH --account=bhurwitz\n",
    "#SBATCH --partition=standard\n",
    "#SBATCH --nodes=1\n",
    "#SBATCH --ntasks=1\n",
    "#SBATCH --time=20:00:00\n",
    "#SBATCH --cpus-per-task=4\n",
    "#SBATCH --mem-per-cpu=1000\n",
    "#SBATCH --array=0-2\n",
    "\n",
    "pwd; hostname; date\n",
    "\n",
    "source ./config.sh\n",
    "names=($(cat $XFILE_DIR/$XFILE))\n",
    "SAMPLE_ID=${names[${SLURM_ARRAY_TASK_ID}]}\n",
    "\n",
    "GENOMAD=${OUT_GENOMAD}/${SAMPLE_ID}/contigs_summary/contigs_virus.fna\n",
    "\n",
    "#load environment\n",
    "CONDA=\"/groups/bhurwitz/miniconda3\"\n",
    "source $CONDA/etc/profile.d/conda.sh\n",
    "conda activate checkv_env\n",
    "\n",
    "checkv end_to_end ${GENOMAD} ${OUT_CHECKV_GENOMAD}/${names[${SLURM_ARRAY_TASK_ID}]} -t 16\n",
    "'''\n",
    "\n",
    "with open('run_checkv_genomad.sh', mode='w') as file:\n",
    "    file.write(my_code)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "32f57747",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses\n",
      "06_get_viruses.ipynb\t   Job-genomad-1.out   out_dvf\n",
      "Job-checkv-dvf-0.out\t   Job-genomad-2.out   out_genomad\n",
      "Job-checkv-dvf-1.out\t   Job-virsort2-0.out  out_virsort\n",
      "Job-checkv-dvf-2.out\t   Job-virsort2-1.out  run_checkv_dvf.sh\n",
      "Job-checkv-virsort2-0.out  Job-virsort2-2.out  run_checkv_genomad.sh\n",
      "Job-checkv-virsort2-1.out  config.sh\t       run_checkv_virsort2.sh\n",
      "Job-checkv-virsort2-2.out  dvf.fasta\t       run_dvf.sh\n",
      "Job-dvf-0.out\t\t   id_from_fasta.py    run_genomad.sh\n",
      "Job-dvf-1.out\t\t   out_checkv_dvf      run_virsort2.sh\n",
      "Job-dvf-2.out\t\t   out_checkv_genomad  temp.sh\n",
      "Job-genomad-0.out\t   out_checkv_virsort\n"
     ]
    }
   ],
   "source": [
    "#check that I am in working directory with config file and run_virsort2.sh script\n",
    "!pwd\n",
    "!ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "56e256d3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Submitted batch job 3298677\r\n"
     ]
    }
   ],
   "source": [
    "#run script!\n",
    "!sbatch ./run_checkv_genomad.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "208bd686",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n",
      "           3298639  standard  jupyter kolodisn  R      41:26      1 i16n4\n"
     ]
    }
   ],
   "source": [
    "#is the job in queue? \n",
    "!squeue --user=$netid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "a713cbad",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_checkv_genomad/ERR2198703:\r\n",
      "complete_genomes.tsv  contamination.tsv  quality_summary.tsv  viruses.fna\r\n",
      "completeness.tsv      proviruses.fna\t tmp\r\n",
      "\r\n",
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_checkv_genomad/ERR2198704:\r\n",
      "complete_genomes.tsv  contamination.tsv  quality_summary.tsv  viruses.fna\r\n",
      "completeness.tsv      proviruses.fna\t tmp\r\n",
      "\r\n",
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses/out_checkv_genomad/ERR2198705:\r\n",
      "complete_genomes.tsv  contamination.tsv  quality_summary.tsv  viruses.fna\r\n",
      "completeness.tsv      proviruses.fna\t tmp\r\n"
     ]
    }
   ],
   "source": [
    "#check the output files \n",
    "!ls $work_dir/out_checkv_genomad/*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "00d2bf90",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses\r\n",
      "i16n12.ocelote.hpc.arizona.edu\r\n",
      "Sun Jul 21 12:24:48 MST 2024\r\n",
      "\r\n",
      "CheckV v0.9.0: contamination\r\n",
      "[1/8] Reading database info...\r\n",
      "[2/8] Reading genome info...\r\n",
      "[3/8] Skipping gene calling...\r\n",
      "[4/8] Reading gene info...\r\n",
      "[5/8] Running hmmsearch...\r\n",
      "[6/8] Annotating genes...\r\n",
      "[7/8] Identifying host regions...\r\n",
      "[8/8] Writing results...\r\n",
      "Run time: 171.94 seconds\r\n",
      "Peak mem: 0.31 GB\r\n",
      "\r\n",
      "CheckV v0.9.0: completeness\r\n",
      "[1/8] Skipping gene calling...\r\n",
      "[2/8] Initializing queries and database...\r\n",
      "[3/8] Running DIAMOND blastp search...\r\n",
      "[4/8] Computing AAI...\r\n",
      "[5/8] Running AAI based completeness estimation...\r\n",
      "[6/8] Running HMM based completeness estimation...\r\n",
      "[7/8] Determining genome copy number...\r\n",
      "[8/8] Writing results...\r\n",
      "Run time: 46.32 seconds\r\n",
      "Peak mem: 1.38 GB\r\n",
      "\r\n",
      "CheckV v0.9.0: complete_genomes\r\n",
      "[1/7] Reading input sequences...\r\n",
      "[2/7] Finding complete proviruses...\r\n",
      "[3/7] Finding direct/inverted terminal repeats...\r\n",
      "[4/7] Filtering terminal repeats...\r\n",
      "[5/7] Checking genome for completeness...\r\n",
      "[6/7] Checking genome for large duplications...\r\n",
      "[7/7] Writing results...\r\n",
      "Run time: 0.08 seconds\r\n",
      "Peak mem: 1.38 GB\r\n",
      "\r\n",
      "CheckV v0.9.0: quality_summary\r\n",
      "[1/6] Reading input sequences...\r\n",
      "[2/6] Reading results from contamination module...\r\n",
      "[3/6] Reading results from completeness module...\r\n",
      "[4/6] Reading results from complete genomes module...\r\n",
      "[5/6] Classifying contigs into quality tiers...\r\n",
      "[6/6] Writing results...\r\n",
      "Run time: 0.05 seconds\r\n",
      "Peak mem: 1.38 GB\r\n",
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses\r\n",
      "i16n12.ocelote.hpc.arizona.edu\r\n",
      "Sun Jul 21 12:24:47 MST 2024\r\n",
      "\r\n",
      "CheckV v0.9.0: contamination\r\n",
      "[1/8] Reading database info...\r\n",
      "[2/8] Reading genome info...\r\n",
      "[3/8] Skipping gene calling...\r\n",
      "[4/8] Reading gene info...\r\n",
      "[5/8] Running hmmsearch...\r\n",
      "[6/8] Annotating genes...\r\n",
      "[7/8] Identifying host regions...\r\n",
      "[8/8] Writing results...\r\n",
      "Run time: 312.45 seconds\r\n",
      "Peak mem: 0.32 GB\r\n",
      "\r\n",
      "CheckV v0.9.0: completeness\r\n",
      "[1/8] Skipping gene calling...\r\n",
      "[2/8] Initializing queries and database...\r\n",
      "[3/8] Running DIAMOND blastp search...\r\n",
      "[4/8] Computing AAI...\r\n",
      "[5/8] Running AAI based completeness estimation...\r\n",
      "[6/8] Running HMM based completeness estimation...\r\n",
      "[7/8] Determining genome copy number...\r\n",
      "[8/8] Writing results...\r\n",
      "Run time: 54.5 seconds\r\n",
      "Peak mem: 1.5 GB\r\n",
      "\r\n",
      "CheckV v0.9.0: complete_genomes\r\n",
      "[1/7] Reading input sequences...\r\n",
      "[2/7] Finding complete proviruses...\r\n",
      "[3/7] Finding direct/inverted terminal repeats...\r\n",
      "[4/7] Filtering terminal repeats...\r\n",
      "[5/7] Checking genome for completeness...\r\n",
      "[6/7] Checking genome for large duplications...\r\n",
      "[7/7] Writing results...\r\n",
      "Run time: 0.12 seconds\r\n",
      "Peak mem: 1.5 GB\r\n",
      "\r\n",
      "CheckV v0.9.0: quality_summary\r\n",
      "[1/6] Reading input sequences...\r\n",
      "[2/6] Reading results from contamination module...\r\n",
      "[3/6] Reading results from completeness module...\r\n",
      "[4/6] Reading results from complete genomes module...\r\n",
      "[5/6] Classifying contigs into quality tiers...\r\n",
      "[6/6] Writing results...\r\n",
      "Run time: 0.08 seconds\r\n",
      "Peak mem: 1.5 GB\r\n",
      "/xdisk/bhurwitz/virus_hunting/kolodisner/06_get_viruses\r\n",
      "i16n13.ocelote.hpc.arizona.edu\r\n",
      "Sun Jul 21 12:24:48 MST 2024\r\n",
      "\r\n",
      "CheckV v0.9.0: contamination\r\n",
      "[1/8] Reading database info...\r\n",
      "[2/8] Reading genome info...\r\n",
      "[3/8] Skipping gene calling...\r\n",
      "[4/8] Reading gene info...\r\n",
      "[5/8] Running hmmsearch...\r\n",
      "[6/8] Annotating genes...\r\n",
      "[7/8] Identifying host regions...\r\n",
      "[8/8] Writing results...\r\n",
      "Run time: 301.51 seconds\r\n",
      "Peak mem: 0.32 GB\r\n",
      "\r\n",
      "CheckV v0.9.0: completeness\r\n",
      "[1/8] Skipping gene calling...\r\n",
      "[2/8] Initializing queries and database...\r\n",
      "[3/8] Running DIAMOND blastp search...\r\n",
      "[4/8] Computing AAI...\r\n",
      "[5/8] Running AAI based completeness estimation...\r\n",
      "[6/8] Running HMM based completeness estimation...\r\n",
      "[7/8] Determining genome copy number...\r\n",
      "[8/8] Writing results...\r\n",
      "Run time: 55.28 seconds\r\n",
      "Peak mem: 1.51 GB\r\n",
      "\r\n",
      "CheckV v0.9.0: complete_genomes\r\n",
      "[1/7] Reading input sequences...\r\n",
      "[2/7] Finding complete proviruses...\r\n",
      "[3/7] Finding direct/inverted terminal repeats...\r\n",
      "[4/7] Filtering terminal repeats...\r\n",
      "[5/7] Checking genome for completeness...\r\n",
      "[6/7] Checking genome for large duplications...\r\n",
      "[7/7] Writing results...\r\n",
      "Run time: 0.12 seconds\r\n",
      "Peak mem: 1.51 GB\r\n",
      "\r\n",
      "CheckV v0.9.0: quality_summary\r\n",
      "[1/6] Reading input sequences...\r\n",
      "[2/6] Reading results from contamination module...\r\n",
      "[3/6] Reading results from completeness module...\r\n",
      "[4/6] Reading results from complete genomes module...\r\n",
      "[5/6] Classifying contigs into quality tiers...\r\n",
      "[6/6] Writing results...\r\n",
      "Run time: 0.08 seconds\r\n",
      "Peak mem: 1.51 GB\r\n"
     ]
    }
   ],
   "source": [
    "#check Job output!\n",
    "!cat Job-checkv-genomad-*.out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "c970a8b9",
   "metadata": {},
   "outputs": [],
   "source": [
    "#copy notebook into current working directory\n",
    "!cp ~/06_get_viruses.ipynb $work_dir"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1d172b4f",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
